plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -193,12 +196,14 @@ public void setFilterPath(File path) {
 
     @Override
     protected boolean doCheckState() {
+      String directory = getStringValue();
       if (!super.doCheckState()) {
-        status = new Status(IStatus.ERROR, getClass().getName(), "Invalid directory");
+        String message = MessageFormat.format(SdkUiMessages.NoSuchDirectory, directory);
```

Comment:
This happens if the user specifies a file, so "No Such Directory" isn't really appropriate.